sintr_worker/lib/basic_info.dart

```diff
@@ -97,3 +97,37 @@ String logKey, Stream<String> stream) async {
   result[basic_info_key] = [unknown_value];
   return result;
 }
+
+Future<Map<String, List<String>>> reduce(
+    String basicKey, Stream<String> stream) async {
+  if (!basicKey.startsWith(key_prefix)) throw 'unknown key: $basicKey';
+  var dateSuffix = basicKey.substring(key_prefix.length);
+  // Map of clientId:clientVersion to set of UUID
+  Map<String, Set<String>> clientMap = {};
+  Map<String, Set<String>> serverMap = {};
+  Map<String, Set<String>> sdkMap = {};
+  await for (String line in stream) {
+    // <uuid>:<clientId>:<clientVersion>:<serverVersion>:<sdkVersion>
+    var split = line.split(':');
```

Comment:
This is surprisingly verbose - as think it's going to be a pattern that we're doing a lot of, is there anything can do to tighten it up? I'm not sure there is obviously, but it would definately be nice